src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -118,7 +118,7 @@ public static void EmitObject(string objectFilePath, IEnumerable<DependencyNode>
 
                     ObjectData nodeContents = node.GetData(factory);
 
-                    dumper?.DumpObjectNode(factory.NameMangler, node, nodeContents);
+                    dumper?.DumpObjectNode(factory, node, nodeContents);
 #if DEBUG
                     foreach (ISymbolNode definedSymbol in nodeContents.DefinedSymbols)
                     {
```

Comment:
yes, that is odd.  Whether I change the test for initialized to 1 or 0 seems to make no difference.  Have changed it to 0 to match x64 and others

Reply:
I think this is ok.  It doesn't seem to make any difference because `CheckStaticClassConstruction` has its own check for 0 so will return immediately anyway.  This is just an earlier check to save that call, but if it is wrong the cctor is not run again anyway.  There is perhaps a small edge problem with using 1 as a marker that another thread is running the cctor, in that 1 is a valid method table index for wasm (as is 0).  Its possible that those indicies are valid `cctorMethodAddress` although in practice I think many other functions are going to be created before the compiler gets to a cctor.